docs/projects/merge-sort/_posts/2019-01-24-python.md

```diff
@@ -0,0 +1,340 @@
+    try:
+        xs = input_list(args[0])
+        if len(xs) <= 1:
+            exit_with_error()
+        print(selection_sort(xs))
+    except (IndexError,ValueError):
+        exit_with_error()
+```
+
+This is the main function of this file. It parses the input, then calls our selection sort
+function (and prints the results). It also deals with any errors raised.
+
+### Transform Input Parameters
+
+```python
+def input_list(list_str):
+    return [int(x.strip(" "), 10) for x in list_str.split(',')]
+```
+
+This function takes a string like `"2, 1, 10, 5, 3"`, and turns into a list of numbers.
+It does this using a list comprehension, first we need to convert our string into a
+list `list_str.split(',')` which is a list of strings split by comma (,).
+So our original input string becomes `["2", " 1", " 10", " 5", " 3"]`. Then for each
+element in the list `for x in ...` ,  we do something to it.
+
+In this example we convert it into a decimal integer, `int(x.strip(" "), 10)`
+`x.strip(" ")`, removes any whitespace so `" 1"` becomes `"1"`, then `int("1", 10)`
+converts the string `"1"` into a decimal number in this case `1`. This is done
+for every item in the list so our original input of `"2, 1, 10, 5, 3"`
+becomes `[2, 1, 10, 5, 3]`.
+
+### Throw Errors
+
+```python
+def exit_with_error():
+    print('Usage: please provide a list of at least two integers to sort in the format “1, 2, 3, 4, 5”')
+    sys.exit(1)
+```
+
+This function prints a message and then exits the script with an error, `sys.exit(1)`.
+If any non-zero value is returned then the program didn't complete properly. This function is called
+if the user input isn't correct.
+
+### Merge
+
+```python
+def merge(xs, ys):
+    if len(xs) <= 0:
+        return ys
+    if len(ys) <= 0:
+        return xs
+    if xs[0] < ys[0]:
+        return [xs[0]] + merge(xs[1:], ys)
+    return [ys[0]] + merge(xs, ys[1:])
+```
+
+This function is used to sort two list together (`xs` and `ys`). There are a different conditions
+we must account for. The first two conditions check if either of the lists are empty. There are
+a few ways to check if a list if empty such as `not xs` but in this case we are checking
+if the length of the list is less than or equal to 0 `len(xs) <= 0`. If either of the
+lists `xs` or `ys`, then we return the non empty list. So if `xs` is empty we will
+return the list `ys`.
+
+The next thing we check for is if the first element of `xs` is less than the `ys`,
+We can get the first element of a list like so `xs[0]`, then we
+`return [xs[0]] + merge(xs[1:], ys)`. This is an example of recursion where the function
+calls itself within the function, in this case the `merge()` function calling the `merge()`
+function again.
+
+Since `xs[0]` is less than `y[0]` we know `xs[0]` must be
+the smallest item in either list, as both lists are already sorted. So we then take `xs[0]` out
+of the first list we convert it into a list `[xs[0]]` because we want to concatenate it with a
```

Comment:
I'm not sure how to clean up this sentence. It's a little clunky to read, so I tried inserting and to break it up a bit.

```suggestion
of the first list, and we convert it into a list `[xs[0]]` because we want to concatenate it with a
```